docs/src/prototypes/dropdowns/inputWithDropdown.tsx

```diff
@@ -0,0 +1,143 @@
+import * as React from 'react'
+import * as ReactDOM from 'react-dom'
+import * as _ from 'lodash'
+import keyboardKey from 'keyboard-key'
+import { Dropdown, DropdownProps, Input } from '@stardust-ui/react'
+
+import { atMentionItems, AtMention } from './dataMocks'
+import { insertNodeAtCursorPosition, removeElement } from './utils'
+
+interface InputWithDropdownState {
+  dropdownOpen?: boolean
+  dropdownValue?: string
+  searchQuery?: string
+}
+
+class InputWithDropdownExample extends React.Component<{}, InputWithDropdownState> {
+  private readonly mountNodeId = 'dropdown-mount-node'
+  private readonly dropdownInputSelector = `#${this.mountNodeId} .${Input.slotClassNames.input}`
+  private readonly editorStyle: React.CSSProperties = {
+    backgroundColor: 'lightgrey',
+    padding: '5px',
+    minHeight: '100px',
+    outline: 0,
+  }
+
+  private contendEditableRef = React.createRef<HTMLDivElement>()
+
+  public state: InputWithDropdownState = {
+    dropdownOpen: false,
+    dropdownValue: null,
+    searchQuery: '',
+  }
+
+  render() {
+    return (
+      <>
+        <div
+          contentEditable
+          ref={this.contendEditableRef}
+          onKeyUp={this.handleEditorKeyUp}
+          style={this.editorStyle}
+        />
+        {this.renderDropdown()}
+      </>
+    )
+  }
+
+  private renderDropdown = () => {
+    if (!this.state.dropdownOpen) {
+      return null
+    }
+
+    insertNodeAtCursorPosition({ id: this.mountNodeId })
+    const node = this.getMountNode()
+
+    if (!node) {
+      return null
+    }
+
+    return ReactDOM.createPortal(
+      <Dropdown
+        defaultOpen={true}
+        inline
+        search
+        items={atMentionItems}
+        toggleIndicator={null}
+        searchInput={{
+          input: { autoFocus: true },
```

Comment:
Can we avoid `setInputElementSize()` and just pass `size` as prop?

```ts
searchInput={{
  input: {
    autoFocus: true,
    size: dropdownOpen ? 0 : searchQuery.length.
  },
}}

Reply:
yup, with the portal approach it's possible, good call; when I was rendering another node it didn't work; also, size needs to be min 1 so I did `searchQuery.length + 1`